src/classes/groundplaces.ts

```diff
@@ -47,13 +54,21 @@ export class GroundPlacesController {
   }
 
   /**
-   * @description Returns a list of places.
+   * @description Returns a list of ground places.
    * @param {string} query - Can be a name, a Gpuid, a unique name or other name.
-   * @param {AutoCompleteFilters} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
+   * @param {AutoCompleteFilters[]|undefined} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
+   *
+   * If you do not give filters, the list will not be filtered.
    * @returns {AutoComplete[]}
    */
   // @ts-ignore
-  public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete[] {}
+  public autocomplete(query: string, filters?: AutoCompleteFilters[]): AutoComplete[] {
+    const groundPlacesFiltered: AutoComplete[] = this.storageService.filterGroundPlaces(filters);
```

Comment:
Logic behind is weird.
1st you search autocomplete then you filtered. and not the opposite.

Reply:
I was wondering what was the best way to do it: search then filter, or first use the filters and search through the filtered results. I still don't see which one has the best impact?

Reply:
it seems to be more efficient to search first because it will really reduce the initial object size and the filter will be much more efficient. if your search is efficient i think it should be better to search then filter